plugins/services/sandbox/controller_service.go

```diff
@@ -224,3 +224,18 @@ func (s *controllerService) Metrics(ctx context.Context, req *api.ControllerMetr
 		Metrics: metrics,
 	}, nil
 }
+
+func (s *controllerService) UpdateResource(
```

Comment:
Hi @abel-von @Burning1020 would you please add more description for this? 

I'm confused. If the `Resource` can represent the task, it seems that it's conflict with existing task API exported by shim V2. Let's say that the task can be added by this new endpoint `UpdateResource`. How to maintain the rest of lifecycle of that task? Is it managed by task API? If yes, I think the issue you run into is about how to expose the task API from sandbox to containerd. If no, I think you're designing new task API. 

Please correct me if my understanding is wrong. Thanks.

Reply:
Actually, This API is only for sandbox controller to add/delete/update resources into a sandbox. not really starting the task. as if we assume that the Task API is running inside the sandbox, it may not update the resources of the sandbox as it is running INSIDE the sandbox. For a sandboxed Task, we call the UpdateResource API to sandbox controller, before we create or delete a task, so that the sandbox controller, which is running outside the sandbox, can update the resource of the sandbox.

Reply:
For example, for the vm based sandbox, the Task API server is running inside the VM, so if we need to start a task in the sandbox, we may have to add more cpus or memories, or hotplug some devices into the VM. this can only be done by sandbox controller, as Task API server is running INSIDE the VM, it can not update the resource of the VM.

Reply:
Synced with @abel-von offline. I think we should rename this api name because it's weird to have delete operation in UpdateResource. It's more like json-patch operations. The sandbox resource has been patch-updated, for example, we should add mount/device for each container. It's worth one more round to discuss this api in weekly meeting.


REF: https://jsonpatch.com/

Reply:
changed to Update with fields